File: 2022/day1CalorieCounting/day1.py
# --- Day 1: Calorie Counting ---

input = "day1Input.txt"

with open(input, "r", encoding="utf-8") as file:
    lines = file.read().split("\n")

bag = {}
elf = 1

for calories in lines:
    if calories == "":
        elf += 1
    else:
        bag[f"calories_elf_{elf}"] = bag.get(f"calories_elf_{elf}", 0) + int(calories)

max_calories_elf = max(bag.values(), default=0)

top_3 = sorted(bag.values())[-3:]

top_3_calories = sum(top_3)

print(f"Calories of the max elf: {max_calories_elf}")  # Answer Part 1

print(f"Calories of the top 3 elfs: {top_3_calories}")  # Answer Part 2

File: 2022/day1CalorieCounting/day1.mjs
// --- Day 1: Calorie Counting ---

import { readFileSync } from "node:fs";

const input = "day1Input.txt";

const lines = readFileSync(input, { encoding: "utf-8" });

const inventory = lines.split("\n");

const bag = {};

let elf = 1;

for (const calories of inventory) {
  if (calories === "") {
    elf += 1;
  } else {
    bag[`caloriesElf${elf}`] =
      (bag[`caloriesElf${elf}`] || 0) + parseInt(calories);
  }
}

const maxCaloriesElf = Math.max(...Object.values(bag));

const top3 = Object.values(bag)
  .sort((a, b) => a - b)
  .slice(-3);

const top3Calories = top3.reduce((a, b) => a + b);

console.log(`Calories of the max elf: ${maxCaloriesElf}`); // Answer Part 1

console.log(`Calories of the top 3 elfs: ${top3Calories}`); // Answer Part 2
